### PR TITLE
Allow comfyui to use images as background

### PR DESCRIPTION
**Pull request by mijuku233**
### How to use an image as a background:

Export a palette profile and edit it.

Set the value of `"bg-img"` to `url('your image path/background image.png')`. 

> Example:`"bg-img": "#202020"` ➡ `"bg-img": "url('Users/Mijuku/Pictures/bg.png')"`.

In addition, setting the value of `CLEAR_BACKGROUND_COLOR` to `transparent` can prevent the background from disappearing due to scaling the canvas.

Finally, import the modified configuration file in the comfyui palette and refresh the browser

屏幕截图 2024-08-22 174630

_If you think it is troublesome to modify the palette configuration file, I have the modified configuration file here. 
[ImgBG.json](https://github.com/user-attachments/files/16709173/ImgBG.json)
Just import the `ImgBG.json` in the comfyui colorpalette, then rename the background image to `bg.png` and place it in the `ComfyUI/web` directory, and refresh the browser to take effect._

### CODE REVIEW

**Review by huchenlei — CHANGES_REQUESTED** [review]
The method of using `bg-img` to replace `background-color` seems a bit off to me. Can you make that `bg-img` an optional field in `colorPalette.ts`? Also the schema in `types/colorPalette.ts` needs an update to reflect the changes here.

**Comment by mijuku233** [comment]
> The method of using `bg-img` to replace `background-color` seems a bit off to me. Can you make that `bg-img` an optional field in `colorPalette.ts`? Also the schema in `types/colorPalette.ts` needs an update to reflect the changes here.

Modified, `bg-img` is set as an optional field.

My coding skills are very limited, if there is anything wrong, please modify it directly.

File: src/assets/css/style.css [hunk 1]
```diff
@@ -38,7 +40,7 @@ body {
   overflow: hidden;
   grid-template-columns: auto 1fr auto;
   grid-template-rows: auto 1fr auto;
-  background-color: var(--bg-color);
+  background: var(--bg-img) no-repeat center /cover;
```
Comment (bjszd) [inline src/assets/css/style.css:43]: With your change the background color is gone for users without an image. But it is possible to keep the background color and the image. That way the color will be shown while the image is loading for those users that set a background image. And for the rest the color will still be there.

```suggestion
  background: var(--bg-color) var(--bg-img) no-repeat center /cover;
```

Comment (bjszd) [reply src/assets/css/style.css:43]: If you make the above change then you won't need to specify hex values in the `bg-img` property, only set it when there's an image url.

Comment (mcmonkey4eva) [reply src/assets/css/style.css:43]: better yet: just do `background: var(--bg-color);`, at which point you can just set `bg-color: "url('Users/Mijuku/Pictures/bg.png') no-repeat center /cover";` and get the intended result without altering anything else?

**Comment by huchenlei** [comment]
> better yet: just do background: var(--bg-color);, at which point you can just set bg-color: "url('Users/Mijuku/Pictures/bg.png') no-repeat center /cover"; and get the intended result without altering anything else?

I think custom background is already achievable with current setup. Closing this for now.

File: src/assets/css/style.css (hunk 1 above)
Comment (mijuku233) [reply src/assets/css/style.css:43]: > better yet: just do `background: var(--bg-color);`, at which point you can just set `bg-color: "url('Users/Mijuku/Pictures/bg.png') no-repeat center /cover";` and get the intended result without altering anything else?

If you only set `var(--bg-color)`, other settings that use `var(--bg-color)` will also be affected.

Setting `background: var(--bg-img)` alone can avoid this problem.

屏幕截图 2024-08-30 111719

屏幕截图 2024-08-30 111818

**Comment by mijuku233** [comment]
> > better yet: just do background: var(--bg-color);, at which point you can just set bg-color: "url('Users/Mijuku/Pictures/bg.png') no-repeat center /cover"; and get the intended result without altering anything else?
> 
> I think custom background is already achievable with current setup. Closing this for now.

To customize the image background, at least change `background-color` to `background`.
But there will be a problem if you customize the background by setting `var(--bg-color)`, as I showed above.
Setting the variable `bg-img` separately can avoid that problem. Here is the modified and more concise code:

https://github.com/Comfy-Org/ComfyUI_frontend/compare/main...mijuku233:ComfyUI_frontend:IMG_BG

image

**Comment by mijuku233** [comment]
Submitted new PR:#698

**Pull request closed without merging**